From e4bd3442c2223802bf8c4a4d868e3b9443c7caf4 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <jouni.malinen@oss.qualcomm.com>
Date: Mon, 18 May 2026 12:13:21 +0300
Subject: [PATCH] MLD: Verify link ID validity in MLE in reconfiguration cases

Ignore or reject invalid link ID value 15 in ML reconfiguration.

Signed-off-by: Jouni Malinen <jouni.malinen@oss.qualcomm.com>
[src/ap/ieee802_11_eht.c part only. The rest is in
 patches/783-MLD-Verify-link-ID-validity-in-MLE-in-reconfiguratio.patch;
 this part is split out because hostapd_parse_link_reconf_req_sta_profile()
 is added by r93-0003.]
---
 src/ap/ieee802_11_eht.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/ap/ieee802_11_eht.c b/src/ap/ieee802_11_eht.c
index 53dbe8f..7f64567 100644
--- a/src/ap/ieee802_11_eht.c
+++ b/src/ap/ieee802_11_eht.c
@@ -2065,6 +2065,8 @@ hostapd_parse_link_reconf_req_sta_profile(struct hostapd_data *hapd,
 
 	link_id = sta_control & EHT_PER_STA_RECONF_CTRL_LINK_ID_MSK;
 	wpa_printf(MSG_DEBUG, "MLD: Per-STA profile for link=%u", link_id);
+	if (link_id >= MAX_NUM_MLD_LINKS)
+		goto out;
 
 	reconf_type_mask =
 		sta_control & EHT_PER_STA_RECONF_CTRL_OP_UPDATE_TYPE_MSK;
-- 
2.34.1

